project-base/storefront/components/Blocks/Banners/BannersSliderItem.tsx

```diff
@@ -1,35 +1,45 @@
 import { desktopFirstSizes } from 'components/Theme/mediaQueries';
 import { ImageSizeFragmentApi, SliderItemFragmentApi } from 'graphql/generated';
 import { useGetWindowSize } from 'hooks/ui/useGetWindowSize';
+import { forwardRef } from 'react';
+import { Loader } from 'components/Basic/Loader/Loader';
 
 type BannersSliderItemProps = {
     item: SliderItemFragmentApi;
 };
 
-export const BannersSliderItem: FC<BannersSliderItemProps> = ({
-    item: { webMainImage, mobileMainImage, link, name },
-}) => {
-    const { width } = useGetWindowSize();
-    const image = width > desktopFirstSizes.tablet ? webMainImage : mobileMainImage;
+export const BannersSliderItem = forwardRef<HTMLAnchorElement, BannersSliderItemProps>(
+    ({ item: { webMainImage, mobileMainImage, link, name } }, ref) => {
+        const { width } = useGetWindowSize();
```

Comment:
Do I understand it correctly that we are still depending on client-side JS here? If so, cannot we rewrite it to pure CSS?

Reply:
Yup, JS is still needed. What exactly do you mean by pure CSS in this case of useGetWindowSize? Only CSS would be possible if we wouldn't have two types of images (mobile and desktop) to show. Otherwise it's messing the DOM.